apis/server/container_bridge.go

```diff
@@ -350,7 +349,32 @@ func (s *Server) logsContainer(ctx context.Context, rw http.ResponseWriter, req
 		return err
 	}
 
-	writeLogStream(ctx, rw, tty, opts, msgCh)
+	var logDetailBuffer bytes.Buffer
+	if opts.Details {
+		c, err := s.ContainerMgr.Get(ctx, name)
+		if err != nil {
+			return err
+		}
+
+		if c.HostConfig.LogConfig.LogOpts != nil {
```

Comment:
I think the detail should be done in `daemon/logger/logmessage.go`, because the we don't want to split the logic into api bridge and daemon.

Reply:
I think it's a different view.  For example, the timestamps, the log printed time, we can think the timestamps as the log's inherent attribute,  even if we don't print it, we also must store it in log msg. But the details is different from it, it just control the print pattern, it can be done in client side and if we not set details, it didn't need to collect it.

Reply:
What'more, if we done in the logmessage.go, it's difficult to get Container config such as env/labels.

Reply:
OK I try to change it

Reply:
Updated